webrtc/videoroom/assets/src/room.ts

```diff
@@ -42,22 +42,41 @@ export class Room {
         onSendMediaEvent: (mediaEvent: SerializedMediaEvent) => {
           this.webrtcChannel.push("mediaEvent", { data: mediaEvent });
         },
-        onTrackAdded: ({ stream, peer, metadata }) => {
-          attachStream(stream, peer.id);
-        },
         onConnectionError: setErrorMessage,
         onJoinSuccess: (peerId, peersInRoom) => {
-          this.peers = peersInRoom;
-          this.peers.forEach((peer) => {
-            addVideoElement(peer.id, peer.metadata.displayName, false);
-          });
-
-          this.updateParticipantsList();
+          try {
+            navigator.mediaDevices
+              .getUserMedia(MEDIA_CONSTRAINTS)
+              .then((stream) => {
+                stream
+                  .getTracks()
+                  .forEach((track) => this.webrtc.addTrack(track, stream));
+
+                addVideoElement(LOCAL_PEER_ID, "Me", true);
+                attachStream(stream, LOCAL_PEER_ID);
+
+                this.peers = peersInRoom;
+                this.peers.forEach((peer) => {
+                  addVideoElement(peer.id, peer.metadata.displayName, false);
+                });
+                this.updateParticipantsList();
+              });
+          } catch (error) {
+            console.error(error);
+            setErrorMessage(
+              "Failed to setup video room, make sure to grant camera and microphone permissions"
+            );
+            throw "error";
+          }
```

Comment:
Would it be possible to put getting localStream in a separate function?
Then we would have
```elixir
    this.localStream = getLocalMedia();
    this.localStream
      .getTracks()
      .forEach((track) => this.webrtc.addTrack(track, this.localStream!));

    addVideoElement(LOCAL_PEER_ID, "Me", true);
    attachStream(this.localStream, LOCAL_PEER_ID);
```